src/platform/backends/qemu/qemu_virtual_machine.cpp

```diff
@@ -380,8 +380,23 @@ void mp::QemuVirtualMachine::shutdown(bool force)
             mpl::log(mpl::Level::debug, vm_name, "No process to kill");
         }
 
-        if (state == State::suspended || mp::backend::instance_image_has_snapshot(desc.image.image_path, suspend_tag))
+        if (state == State::suspended && !mp::backend::instance_image_has_snapshot(desc.image.image_path, suspend_tag))
```

Comment:
You can put this into an else block of the below conditional to avoid calling qemu-img unnecessarily.